Source/objects.cpp

```diff
@@ -4590,36 +4250,10 @@ void MonstCheckDoors(const Monster &monster)
 
 		Object &door = *object;
 		// Doors use _oVar4 to track open/closed state, non-zero values indicate an open door
-		// we don't use Object::isDoor since the later conditions will check the object is actually a door anyway.
-		if (door._oVar4 != DOOR_CLOSED)
+		if (!door.isDoor() || door._oVar4 != DOOR_CLOSED)
 			continue;
 
-		// Strictly speaking these checks shouldn't be required, they're guarding against monsters
-		// standing in a wall opening an adjacent door. Shouldn't be possible?
-		if (IsNoneOf(dir, Direction::NorthEast, Direction::SouthWest)) {
-			if (door._otype == OBJ_L1LDOOR) {
-				OperateL1LDoor(door, true);
-			} else if (door._otype == OBJ_L2LDOOR) {
-				OperateL2LDoor(door, true);
-			} else if (door._otype == OBJ_L3RDOOR) {
-				// L3 doors are backwards, see OperateDoor
-				OperateL3RDoor(door, true);
-			} else if (door._otype == OBJ_L5LDOOR) {
-				OperateL5LDoor(door, true);
-			}
-		}
-		if (IsNoneOf(dir, Direction::NorthWest, Direction::SouthEast)) {
-			if (door._otype == OBJ_L1RDOOR) {
-				OperateL1RDoor(door, true);
-			} else if (door._otype == OBJ_L2RDOOR) {
-				OperateL2RDoor(door, true);
-			} else if (door._otype == OBJ_L3LDOOR) {
-				// L3 doors are backwards, see OperateDoor
-				OperateL3LDoor(door, true);
-			} else if (door._otype == OBJ_L5RDOOR) {
-				OperateL5RDoor(door, true);
-			}
-		}
+		OperateDoor(door, true);

```

Comment:
As an aside the only reason this loop operated in the specific order was to allow for branch prediction on the conditions that have now been removed. Could reorder that to go from `Direction::South` to `Direction::SouthEast` in a future cleanup.